homework/index.js

```diff
@@ -36,11 +36,115 @@
       if (err) {
         createAndAppend('div', root, { text: err.message, class: 'alert-error' });
       } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        // create the basic element tags //
+        // the main header.
+        createAndAppend('h1', root, { text: 'HYF Repositories' });
+
+        // the container
+        createAndAppend('div', root, { id: 'container' });
+        const container = document.getElementById('container');
```

Comment:
Hi Bassam, if you check the `createAndAppend` function you will see that it returns the new element which is created. Thus, instead of calling that function and executing `document.getElementById` in the next line, you can try replacing it with a single statement i.e

```
const container = createAndAppend('div', root, { id: 'container' });
```

This saves you one line of code but more importantly, you don't have to search for the element in document and hence can save some JavaScript execution time.